netmiko/fortinet/fortinet_ssh.py

```diff
@@ -1,104 +1,298 @@
+		)
+
+	def disable_paging(
+		self,
+		command: str = "",
+		delay_factor: Optional[float] = None,
+		cmd_verify: bool = True,
+		pattern: Optional[str] = None,
+		) -> str:
+		"""Disable paging is only available with specific roles so it may fail.
+
+		:param command: Device command to disable pagination of output
+
+		:param delay_factor: Deprecated in Netmiko 4.x. Will be eliminated in Netmiko 5.
+		"""
+
+		disable_paging_commands = [
+			"config global",
+			"config system console",
+			"set output standard",
+			"end"
+			]
+		output = self.send_config_set(
+			config_commands = disable_paging_commands,
+			exit_config_mode = True,
+			read_timeout = None,
+			delay_factor = 1.0,
+			max_loops = 150,
+			enter_config_mode = False,
+		)
+		log.debug("***Window Paging Disabled***")
+		return output
+
+	def cleanup(self, command: str = "quit") -> None:
+		"""Re-enable paging globally."""
+		# Return paging state
+		enable_paging_commands = ["config global", "config system console", "set output more", "end"]
+		# Should test output is valid
+		output = self.send_config_set(
+			config_commands = enable_paging_commands,
+			exit_config_mode = True,
+			read_timeout = None,
+			delay_factor = 1.0,
+			max_loops = 150,
+			enter_config_mode = False,
+		)
+		log.debug("***Window Paging Enabled***")
+
+		"""Gracefully exit the SSH session."""
+		try:
+			self.exit_config_mode()
+		except Exception:
+			pass
+		# Always try to send final 'exit' (command)
+		self._session_log_fin = True
+		self.write_channel(command + self.RETURN)
+
+	def save_config(
+		self, cmd: str = "", confirm: bool = False, confirm_response: str = ""
+		) -> str:
+		"""Not Implemented, platform automatically saves"""
+		raise NotImplementedError
+
+	def _enter_global_or_vdom(self, mode_name: str = None):
+		"""This allows a device enter global or vdom mode and confirm before proceeding.
+
+		:param mode_name: the name of the mode trying to be entered: 'global', 'root' or the name of the vdom string.
+		"""
+		if mode_name == "global":
+			log.debug("***Entering Global Mode***")
+			try: 
+				self.send_command(command_string="config global", expect_string=r"\(global\) # ")
+			except:
+				raise ValueError("Device Failed to Enter Global Mode.")
+		else:
+			log.debug("***Entering VDOM Mode***")
+			try:
+				self.send_command(command_string="config vdom", expect_string=r"\(vdom\) # ")
+				self.send_command(command_string=f"edit {mode_name:}", expect_string=fr"\({mode_name}\) # ")
+			except:
+				raise ValueError(f"Device Failed to Enter '{mode_name}' VDOM Mode.")
+
+	def _exit_global_or_vdom(self):
+		"""This allows a device to exit global or vdom mode and confirm before proceeding"""
+		mode_active = True
+		log.debug("***Attempting to Exit Mode***")
+		loops = 0
+
+		while mode_active:
+			output = self.find_prompt()
+
+			if ") #" in output:
+				self.write_channel("end")
+			elif ") #" not in output or loops == 5:
+				break
+
+			time.sleep(1)
+			loops += 1
```

Comment:
Once again probably should be two separate methods.

Reply:
If you separate them, it would be kind of redundant here as it's the same code, but if adds ease of use, I'm fine with that.